### PR TITLE
Email documentation

### PR DESCRIPTION
**Pull request by joshmoore**
This documents the work from:

 * https://github.com/openmicroscopy/openmicroscopy/pull/3542
 * https://github.com/openmicroscopy/openmicroscopy/pull/3434

and aims to document all the various components (user, dev, sysadmin) for the OMERO.mail subsystem.

### CODE REVIEW

**Comment by hflynn** [comment]
--no-rebase

**Comment by mtbc** [comment]
At a glance, this looks pretty good.

File: omero/sysadmins/mail.txt [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+property will need to be activated.
+
+Minimum configuration
+---------------------
+
+::
+
+    $ bin/omero config set omero.mail.config true
+
+By default, this will use "localhost" as the mail server on port 25 and send as
```
Comment (manics) [inline omero/sysadmins/mail.txt]: Is there a sphinx role for in-line config values ("localhost", "omero")?

File: omero/sysadmins/mail.txt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+events on the server. Those available include:
+
+* `ServerUpMailSender` and `ServerDownMailSender` which mail when the server goes up or down
+* `FailedLoginMailSender` which can be configured to send for particular users if a bad password is used
+* `ObjectMailSender` which can be configured to send an email under various conditions. Instances which are configured include:
+   * `newUserMailSender` which sends an email every time a user is created
+   * `newCommentMailSender` which sends an email every time a user's image is commented on by another user.
+
+To activate the senders, the :file:`etc/blitz/mail-senders.example` can be copied
+to a file ending with ".xml".
```
Comment (manics) [inline omero/sysadmins/mail.txt]: Copied to `etc/blitz/mail-senders.xml`. Server down email worked, Login failure results in a Blitz error:
```
2015-03-18 09:40:48,288 ERROR [  ome.security.auth.JdbcPasswordProvider] (.Server-13) LoginAttemptMessage error
java.lang.NullPointerException: null
        at ome.services.mail.FailedLoginMailSender.onApplicationEvent(FailedLoginMailSender.java:75) ~[server.jar:na]
```

**Comment by manics** [comment]
Otherwise looks good

File: omero/sysadmins/mail.txt (hunk 1 above)
Comment (hflynn) [inline omero/sysadmins/mail.txt]: Looks like single back ticks are what's used elsewhere, there is no specific role for config values only commands and environment variables.

File: omero/sysadmins/mail.txt (hunk 2 above)
Comment (joshmoore) [inline omero/sysadmins/mail.txt]: Card filed on devx for 5.1.0.

**Comment by hflynn** [comment]
Is this good to merge @joshmoore @manics ?

File: omero/sysadmins/mail.txt (hunk 1 above)
Comment (manics) [inline omero/sysadmins/mail.txt]: @hflynn I'll let you decide on the best formatting.

**Comment by manics** [comment]
If you and Josh are happy then merge.

File: omero/sysadmins/mail.txt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+Other key properties
+--------------------
+
+Along with :property:`omero.mail.host`, a few general connection properties may
+be needed for your particular SMTP server:
+
+ * :property:`omero.mail.port`
+ * :property:`omero.mail.smtp.auth`
+ * :property:`omero.mail.smtp.starttls.enable`
```
Comment (hflynn) [inline omero/sysadmins/mail.txt]: These bullet points need moving back against the margin, getting that annoying highlighting block in plone.

**Comment by hflynn** [comment]
I'll fix the highlighting when I do the rest of the what's new etc updates.

**Pull request merged**